File: social-recipe-app-main/frontend/src/services/mealPlan.service.ts
import api from './api';

export interface MealPlan {
  id: number;
  recipeId: number;
  recipeTitle: string;
  recipeImageUrl: string;
  plannedDate: string;
  mealType: string;
  servingsAdjustment: number;
  status: string;
  calories?: number;
  protein?: number;
  carbs?: number;
  fats?: number;
}

export interface MealPlanRequest {
  recipeId: number;
  plannedDate: string;
  mealType: string;
  servingsAdjustment?: number;
  status?: string;
}

export const mealPlanService = {
  getMealPlans: (startDate: string, endDate: string) =>
    api.get<MealPlan[]>(`/meal-planner`, { params: { startDate, endDate } }).then(r => r.data),
  
  addMealPlan: (request: MealPlanRequest) =>
    api.post<MealPlan>(`/meal-planner`, request).then(r => r.data),
  
  deleteMealPlan: (id: number) =>
    api.delete(`/meal-planner/${id}`).then(r => r.data),

  updateMealPlan: (id: number, request: Partial<MealPlanRequest>) =>
    api.put<MealPlan>(`/meal-planner/${id}`, request).then(r => r.data),
};

File: social-recipe-app-main/backend/src/main/java/com/bluepal/service/interfaces/MealPlanService.java
package com.bluepal.service.interfaces;

import com.bluepal.dto.request.MealPlanRequest;
import com.bluepal.dto.response.MealPlanResponse;
import com.bluepal.entity.User;

import java.time.LocalDate;
import java.util.List;

public interface MealPlanService {
    MealPlanResponse addMealPlan(User user, MealPlanRequest request);
    List<MealPlanResponse> getMealPlans(User user, LocalDate startDate, LocalDate endDate);
    MealPlanResponse updateMealPlan(Long id, MealPlanRequest request, User user);
    void deleteMealPlan(Long id, User user);
}
